src/App.tsx

```diff
@@ -17,18 +48,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                setFilter={setFilter}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 ? (
+                <Loader />
```

Comment:
The length of todos being zero does not necessarily mean data is still loading. It could also mean that there are no todos. A separate loading state is recommended to differentiate these two cases.